src/librustc_resolve/resolve_imports.rs

```diff
@@ -343,6 +343,25 @@ struct ImportResolver<'a, 'b: 'a> {
     resolver: &'a mut Resolver<'b>,
 }
 
+impl<'a, 'b: 'a> ::std::ops::Deref for ImportResolver<'a, 'b> {
+    type Target = Resolver<'b>;
+    fn deref(&self) -> &Resolver<'b> {
+        self.resolver
+    }
+}
```

Comment:
I feel dirty just reviewing this code - I really wish we didn't use Deref like this in name resolution. Still the damage is done, I guess this doesn't make things much worse.